scaler/io/ymq/ymq.py

```diff
@@ -0,0 +1,101 @@
+# This file wraps the interface exported by the C implementation of the module
+# and provides a more ergonomic interface supporting both asynchronous and synchronous execution
+
+__all__ = ["IOSocket", "IOContext", "Message", "IOSocketType", "YMQException", "Bytes", "ErrorCode"]
+
+import asyncio
+import concurrent.futures
+from typing import Optional
+
+from scaler.io.ymq._ymq import BaseIOSocket, Message, IOSocketType, BaseIOContext, YMQException, Bytes, ErrorCode
+
+async def call_async(func, *args, **kwargs):
+    future = asyncio.get_event_loop().create_future()
+    def callback(result):
```

Comment:
```suggestion
P = ParamSpec("P")
T = TypeVar("T")

async def call_async(func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs) -> T:
    future = asyncio.get_event_loop().create_future()
    def callback(result: Union[T, Exception]) -> None:
```

I think this would work with all functions as the callback's `result: Union[T, Exception]` matches both `Optional[Exception]` (e.g. `send()`) and `Union[Message, Exception]` (`recv`).